airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -73,191 +73,129 @@
-
-            final SyncOutput syncWorkflowInputs = runMandatoryActivityWithOutput(
-                (input) -> getSyncInputActivity.getSyncWorkflowInput(input),
-                getSyncInputActivitySyncInput);
-
-            workflowState.setRunning(true);
-
-            final SyncWorkflow childSync = Workflow.newChildWorkflowStub(SyncWorkflow.class,
-                ChildWorkflowOptions.newBuilder()
-                    .setWorkflowId("sync_" + maybeJobId.get())
-                    .setTaskQueue(TemporalJobType.CONNECTION_UPDATER.name())
-                    // This will cancel the child workflow when the parent is terminated
-                    .setParentClosePolicy(ParentClosePolicy.PARENT_CLOSE_POLICY_TERMINATE)
-                    .build());
-
-            final UUID connectionId = connectionUpdaterInput.getConnectionId();
-
-            try {
-              standardSyncOutput = Optional.ofNullable(childSync.run(
-                  syncWorkflowInputs.getJobRunConfig(),
-                  syncWorkflowInputs.getSourceLauncherConfig(),
-                  syncWorkflowInputs.getDestinationLauncherConfig(),
-                  syncWorkflowInputs.getSyncInput(),
-                  connectionId));
-
-              final StandardSyncSummary standardSyncSummary = standardSyncOutput.get().getStandardSyncSummary();
-
-              if (workflowState.isResetConnection()) {
-                workflowState.setResetConnection(false);
-              }
-
-              if (standardSyncSummary != null && standardSyncSummary.getStatus() == ReplicationStatus.FAILED) {
-                failures.addAll(standardSyncOutput.get().getFailures());
-                partialSuccess = standardSyncSummary.getTotalStats().getRecordsCommitted() > 0;
-                workflowState.setFailed(true);
-              }
-            } catch (final ChildWorkflowFailure childWorkflowFailure) {
-              if (childWorkflowFailure.getCause() instanceof CanceledFailure) {
-                // do nothing, cancellation handled by cancellationScope
-
-              } else if (childWorkflowFailure.getCause() instanceof ActivityFailure) {
-                final ActivityFailure af = (ActivityFailure) childWorkflowFailure.getCause();
-                failures.add(FailureHelper.failureReasonFromWorkflowAndActivity(
-                    childWorkflowFailure.getWorkflowType(),
-                    af.getActivityType(),
-                    af.getCause(),
-                    maybeJobId.get(),
-                    maybeAttemptId.get()));
-                throw childWorkflowFailure;
-              } else {
-                failures.add(
-                    FailureHelper.unknownOriginFailure(childWorkflowFailure.getCause(), maybeJobId.get(), maybeAttemptId.get()));
-                throw childWorkflowFailure;
-              }
-            }
-          }
-        });
-        syncWorkflowCancellationScope.run();
+        generateSyncWorkflowRunnable(connectionUpdaterInput);
```

Comment:
```suggestion
        cancellableSyncWorkflow = generateSyncWorkflowRunnable(connectionUpdaterInput);
```

Can we have this return and set `cancellableSyncWorkflow`? Otherwise it isn't clear exactly what's it's doing here without looking at the implementation.

Reply:
Done